plugins/plugin-chart-table/src/transformProps.ts

```diff
@@ -47,25 +48,37 @@ function isTimeColumn(key: string) {
   return key === TIME_COLUMN;
 }
 
+const REGEXP_DATETIME = /^\d{4}-[01]\d-[03]\d/;
 const REGEXP_TIMESTAMP_NO_TIMEZONE = /T(\d{2}:){2}\d{2}$/;
 function isTimeType(key: string, data: DataRecord[] = []) {
-  return isTimeColumn(key) || data.some(x => x[key] instanceof Date);
+  return (
+    isTimeColumn(key) ||
+    data.some(x => {
+      const value = x[key];
+      return value instanceof Date || (typeof value === 'string' && REGEXP_DATETIME.test(value));
+    })
+  );
+}
+
+function isNumeric(key: string, data: DataRecord[] = []) {
+  return data.every(x => x[key] === null || x[key] === undefined || typeof x[key] === 'number');
```

Comment:
Usually we do `typeof x[key] === 'undefined'` to be safe because `undefined` is not a reserved keyword and can be overwritten. 

This is one of the common questions in our codebase as the `typeof xxx === 'undefined'` is kinda verbose. The intention was to better be safe than sorry. 

cc @williaster who introduced me to this caveat. 

---

Good to know in ECMAScript 5.1+ `undefined` was made readonly. There are still small possibility of overwriting via local variable. Although that would be a harder accident to make. (ps. the author of this post seem to be quite mad at this syntax lol)
https://stackoverflow.com/questions/8783510/how-dangerous-is-it-in-javascript-really-to-assume-undefined-is-not-overwritte

Reply:
![image](https://user-images.githubusercontent.com/335541/86420179-9d5f5200-bc8a-11ea-8a5a-c6403e763e27.png)

It seems it can't be overridden in Chrome at least. (Chrome just silently ignore your assignment.)

---

"if they did it would be reason enough to never use any code they wrote... right?" LOL on this line

Reply:
I'm basically fine with anything that ESLint doesn't complaint.